sys/crypto/aes.c

```diff
@@ -1037,7 +1038,7 @@ int aes_encrypt(const cipher_context_t *context, const uint8_t *plainBlock,
     const AES_KEY *key = &aeskey;
 
     res = aes_set_encrypt_key((unsigned char *)context->context,
-                              AES_KEY_SIZE * 8, &aeskey);
+                              context->key_size * 8, &aeskey);
```

Comment:
What if `context->key_size` is not initialized? This happens easily when a user does not utilize the `ciphers` module (see Fortuna RNG). Wouldn't be necessary with the old `cipher_interface_t` struct.